parquet-hadoop/src/main/java/org/apache/parquet/filter2/statisticslevel/StatisticsFilter.java

```diff
@@ -129,8 +139,9 @@ public <T extends Comparable<T>> Boolean visit(Eq<T> eq) {
         return BLOCK_MIGHT_MATCH;
       }
       // we are looking for records where v eq(null)
-      // so drop if there are no nulls in this chunk
-      return !hasNulls(meta);
+      // so if there are no nulls in this chunk, we can drop it,
+      // if there has nulls in this chunk, we must take it
+      return !hasNulls(meta) ? BLOCK_CANNOT_MATCH : BLOCK_MUST_MATCH;
```

Comment:
I suspect that hasNull will always return precise information. When null_count is missing, hasNulls also returns true. Replacing `BLOCK_MUST_MATCH` with `BLOCK_MIGHT_MATCH` here makes more sense to me.